mtls-northbound/README.md

```diff
@@ -0,0 +1,603 @@
+EOF
+```
+Create the strict policy.
+```
+gcloud beta network-security server-tls-policies import ${ROOT}-strict \
+    --source=${ROOT}-strict.yaml \
+    --location=global
+```
+
+## Prepare Configuration files to Update the GLB
+The steps are the same for any Application GLB, just find the Target HTTPS Proxy for the GLB and update.
+You can easily switch between the TLS policies by creating representations for each configuration.
+
+Find the Target HTTPS Proxy for your GLB.
+```
+gcloud compute target-https-proxies list
+                                SSL_CERTIFICATES            URL_MAP              REGION    CERTIFICATE_MAP
+apigee-proxy-https-proxy        gm-xapi-kurtkanaskie-net    apigee-proxy-url-map
+apigee-proxy-modern-https-proxy gm-m-xapi-kurtkanaskie-net  apigee-proxy-modern-url-map
+```
+
+### Create Target HTTPS Proxy configurations
+We'll create multiple configurations so we can easily switch between them, one for "none", "lenient" and "strict".
+
+Export the Target HTTPS Proxy configuration and store in a "none" file.
+This allows us to remove the Server TLS Policy later.
+```
+export TARGET_PROXY=apigee-proxy-https-proxy
+
+gcloud compute target-https-proxies export ${TARGET_PROXY} \
+   --global \
+   --destination=${TARGET_PROXY}-none.yaml
+```
+Copy to create new files for "lenient" and "strict" and add the appropriate Server TLS Policy names.
+
+Lenient
+```
+cp ${TARGET_PROXY}-none.yaml ${TARGET_PROXY}-lenient.yaml
+
+echo "serverTlsPolicy: //networksecurity.googleapis.com/projects/${PROJECT}/locations/global/serverTlsPolicies/${ROOT}-lenient" >> ${TARGET_PROXY}-lenient.yaml
+```
+
+Strict
+```
+cp ${TARGET_PROXY}-none.yaml ${TARGET_PROXY}-strict.yaml
+
+echo "serverTlsPolicy: //networksecurity.googleapis.com/projects/${PROJECT}/locations/global/serverTlsPolicies/${ROOT}-strict" >> ${TARGET_PROXY}-strict.yaml
+```
+
+## Update the GLB Backend Service with mTLS headers
+First, find the Backend Service(s) for the GLB.
+The custom header configuration must be added to each Backend Service, in this example we only have one.
+For example:
+```
+gcloud compute backend-services list --global
+NAME                  BACKENDS                                                                                PROTOCOL
+apigee-proxy-backend  us-east1/instanceGroups/apigee-mig-us-east1,us-west1/instanceGroups/apigee-mig-us-west1 HTTPS
+```
+[Update the custom headers](https://cloud.google.com/load-balancing/docs/https/custom-headers-global#mtls-variables).
+We'll also turn on logging so we can check Cloud Logging for errors in "strict" mode.
+
+**CAUTION**: Check if you already have custom headers configured so as not to overwrite them. Use:
+```
+gcloud compute backend-services describe $BACKEND_SERVICE --global --format=json > $BACKEND_SERVICE.json
+```
+
+**NOTE**: There's a limit of 16 custom headers.
+```
+export BACKEND_SERVICE=apigee-proxy-backend
```

Comment:
This seems to belong above line 220.  Also let's make it clear they need to set it to the value returned in their project